### PR TITLE
Multiselect: Enable configuration at instantiation

### PR DESCRIPTION
**Pull request by meissadia**
These changes are in support of development for `design-system-react`.  In order to integrate the DS Multiselect into our React implementation, we need to be able to apply configuration for: 
- Max number of selections (number)
- Should display Tag/Pill (boolean)

## Additions

- feat: Support configuration options for Atomic components
- feat: [Mulitselect] Make rendering of selected Tags configurable
- feat: [Multiselect] Make maximum number of selections configurable

## Pending
- Should we support configuration per-Multiselect when calling `instantiateAll()`? Currently it takes a single configuration object but this could be updated to accept an array of configurations. 

## Testing

In parallel I'm integrating this component into the Design System React and these proposed changes are working great!

### CODE REVIEW

File: packages/cfpb-atomic-component/src/utilities/atomic-helpers.js [hunk 1]
```diff
@@ -115,7 +116,7 @@ function instantiateAll(selector, Constructor, scope) {
     element = elements[i];
     if (contains(element, INIT_FLAG) === false) {
       inst = new Constructor(element);
-      inst.init();
+      inst.init(config);
```
Comment (anselmbradford) [inline packages/cfpb-atomic-component/src/utilities/atomic-helpers.js:119]: Thank you for this! This got me thinking about the architectural approach of configuring components at initialization.

On the one hand it's nice to have a call to `instantiateAll` always return the same thing, without any additional arbitrary config. However, we do need a way to specify different initial states of a component in some cases, such as the config you have in this PR. 

I looked at how we do this with Expandables. In Expandables, there is an option where an Expandable can be open on page load. For this, we add a CSS class [`o-expandable__onload-open`](https://github.com/cfpb/design-system/blob/fb976fcacecc7e5439b822e022be96befe1cbd6a/packages/cfpb-expandables/src/Expandable.js#L49). This class itself adds no styles directly (using JS to set additional CSS that sets the appearance). Per BEM naming, the `o-expandable` sets a certain appearance, and `o-expandable__onload-open` is a modifier on that base appearance. 

The trade offs of this are:

- Since the flag for this configuration appears in the markup, the site can be searched using the crawler for `o-expandable__onload-open` to find all expandables that are open on load. 
- This is possibly an abuse of BEM, since the modifier, `o-expandable_onload-open`, does not actually change the appearance directly itself.

Now, in regard to the multiselect, I could see the max selections being an attribute on the HTML, in the same way the native `<select>` element has a `size` attribute, for example. However, since the native HTML has a limited set of attributes, we would either need to add a custom `data-max-selections` attribute, or take a class-based approach, as we do with expandables. However, neither of these approaches are elegant. I think in the future, the multiselect will probably be a web component and we would set an attribute on the custom element.

So, in conclusion, I think this config approach on `instantiateAll` is fine, but two things should happen in the future through someone's effort:

- Expandables should align better with this same pattern.
- The configuration options should likely be set through some sort of attribute that appears in the markup.

File: packages/cfpb-atomic-component/src/utilities/atomic-helpers.js [hunk 2]
```diff
@@ -102,10 +102,11 @@ function setInitFlag(element) {
  * @param {string} selector - Selector to search for in the document.
  * @param {Function} Constructor - A constructor function.
  * @param {HTMLElement} [scope] - A dom node in which to query the selector.
+ * @param {object} config - Configuration will be provided to the Constructor's init()
  *   If not supplied, it defaults to the `document`.
  * @returns {Array} List of instances that were instantiated.
  */
-function instantiateAll(selector, Constructor, scope) {
+function instantiateAll(selector, Constructor, scope, config = {}) {
```
Comment (anselmbradford) [inline packages/cfpb-atomic-component/src/utilities/atomic-helpers.js:109]: You'll want to update the associated unit tests for this change.

File: packages/cfpb-forms/src/organisms/Multiselect.js [hunk 3]
```diff
@@ -27,6 +27,12 @@ const KEY_UP = 'ArrowUp';
 const KEY_DOWN = 'ArrowDown';
 const KEY_TAB = 'Tab';
 
+// Configuration default
+const DEFAULT_CONFIG = {
+  renderTags: true, // Allow the Multiselect to generate the Tag elements in the DOM
```
Comment (anselmbradford) [inline packages/cfpb-forms/src/organisms/Multiselect.js:32]: In what case would we want this to be false? Is this to support https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/ ? Could we maybe change that to show tags with the multiselect?

Comment (meissadia) [reply packages/cfpb-forms/src/organisms/Multiselect.js:32]: @anselmbradford This option was added to allow for a workaround (we render the Tags ourselves) to a bug we encountered while trying to integrate the DS Multiselect into the `design-system-react` implementation of the DS.  The DS icons are not rendering correctly. 
 
I've tried to debug but have yet to find a resolution. 

|DS Rendered Tags|React rendered Tags|
|---|---|
|Screenshot 2023-11-13 at 11 01 21 AM|Screenshot 2023-11-13 at 11 08 25 AM|

File: packages/cfpb-atomic-component/src/utilities/atomic-helpers.js (hunk 2 above)
Comment (meissadia) [reply packages/cfpb-atomic-component/src/utilities/atomic-helpers.js:109]: Added unit tests for configuration support to both Multiselect and instantiateAll

File: test/unit-test/src/cfpb-atomic-component/src/utilities/atomic-helpers.spec.js [hunk 4]
```diff
@@ -34,6 +35,12 @@ const HTML_SNIPPET = `
           </p>
       </div>
   </div>
+  <select name="test-select" id="test-select" class='o-multiselect' multiple>
```
[the remark this answers is not in the document]
Comment (meissadia) [reply test/unit-test/src/cfpb-atomic-component/src/utilities/atomic-helpers.spec.js:38]: This element is within an automated unit test and not exposed to end users.

**Comment by meissadia** [comment]
The failing browser test for Multiselect is due to a pop-up in the Sauce CI's browser and not due to any actual UI issue. 
 
I'm going to trigger a re-run and see if that gets things to pass, but I hope that false-negative doesn't block this PR. 
 
Video: https://app.saucelabs.com/tests/cb07f176fdbe4634a9d7b2a318b06eca?auth=cf927227628650267280588ab0fa1e63#80

File: packages/cfpb-forms/src/organisms/Multiselect.js (hunk 3 above)
Comment (anselmbradford) [reply packages/cfpb-forms/src/organisms/Multiselect.js:32]: Maybe add a `TODO: ` comment above this explaining why it's needed.

**Review by anselmbradford — APPROVED** [review]
Run `yarn lint` for good measure

**Comment by meissadia** [comment]
> Run `yarn lint` for good measure

I've commited the changes made by `yarn lint`.  
 
I was not able to get the `yarn eslint ...` portion of the linting script to complete successfully.

**Comment by meissadia** [comment]
Thank you very, very much for your guidance and feedback on this @anselmbradford!

File: docs/pages/links.md [hunk 5]
```diff
@@ -198,16 +205,16 @@ variation_groups:
           appear in the following style and include both the original link text
           as well as a shortened URL.
         variation_code_snippet: Here's the <a href="#">link style</a> when printed.
-        variation_code_snippet_rendered: "<p>Here's the <a
-          href=\"https://consumerfinance.gov/about-us/blog\"
-          style=\"font-weight: 500;\">link style<span style=\"border-bottom: 1px
-          solid #ffffff; font-weight: 300;\">
-          (cfpb.gov/about-us/blog)</span></a> when printed.</p>"
+        variation_code_snippet_rendered: '<p>Here''s the <a
```
Comment (meissadia) [inline docs/pages/links.md:208]: These single quotes look iffy but everything is rendering as expected locally, so I'm not going to make any manual adjustments.
